docs/testing.md

```diff
@@ -0,0 +1,39 @@
+# Testing with Shescape
+
+This document provides an overview of why and how to use Shescape's testing
+utilities in your tests.
+
+Please [open an issue] if you found a mistake or if you have a suggestion for
+how to improve the documentation.
+
+## Why
+
+The behaviour of Shescape depends on external factors such as the operating
+system it is running on and environment variables. This may not be desirable in
+your tests, especially in unit tests.
+
+To avoid unexpected behavior you can use the [test stub]s provided by Shescape
+as a drop-in replacement during testing.
+
+## How
```

Comment:
Add a note that Shescape shouldn't be stubbed if it's being used to escape inputs to a command invocation:

```suggestion
as a drop-in replacement during testing.

> **Warning**: If the code under test invokes a command you should **not** use
> these stubs.

## How
```